core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -169,6 +170,19 @@ public Expression visitRelevanceFieldList(RelevanceFieldList node, AnalysisConte
         ImmutableMap.copyOf(node.getFieldList())));
   }
 
+  @Override
+  public Expression visitConstantFunction(ConstantFunction node, AnalysisContext context) {
+    var valueName = node.getFuncName();
+    if (context.getConstantFunctionValues().containsKey(valueName)) {
+      return context.getConstantFunctionValues().get(valueName);
+    }
+
+    var value = visitFunction(node, context);
+    value = DSL.literal(value.valueOf(null));
+    context.getConstantFunctionValues().put(valueName, value);
```

Comment:
Just to confirm, the result of first constant function call will be "cached" and reused by later same function call. So `now()` and `curdate()` may be different, right?

Reply:
That is correct.  `sysdate` returns the date when the `sysdate` _operation_ is executed. `now` returns the date at the start of execution of the _query_.